### PR TITLE
chore(.cursor): add frontend code review skill for cursor

### PR DESCRIPTION
**Pull request by koji**
# Overview
add frontend code review skill for cursor

close AUTH-2881

## Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [ ] update checklist
- [ ]

File: .cursor/skills/grill-frontend-code/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+name: grill-frontend-code
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/SKILL.md:2]: Love some 4th of July grilling 🌭 🌽

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+Open it and check file naming, class naming, design-token usage, and Stylelint rules
+against that file. Do not restate these rules here — they will drift out of sync.
+
+## 3. Web Accessibility (a11y)
+
+Opentrons interfaces are used in physical lab environments. Accessibility is functional reliability.
+
+- **Interactivity**: Any clickable element that is not a semantic `<button>` or `<a>` must be rejected. Do not allow `onClick` on a raw `<div>` without appropriate `role` and `tabIndex`.
+- **Visual Labels**: Icon-only buttons (e.g., an X close button) must explicitly declare an `aria-label` or `aria-describedby`.
+- **Dynamic States**: Loading indicators must possess `aria-busy="true"`.
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:24]: I think `aria-busy` is supposed to be on the container whose elements are being loaded, not on the loading indicator itself.

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## 4. TypeScript & Logic Rigor
+
+Canonical source: `.cursor/skills/opentrons-typescript/SKILL.md` § "Common Pitfalls" and
+"TypeScript Configuration". Cross-check `any` usage, nullability checks, and nested-ternary
+usage against that list.
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:30]: Can we avoid mentioning these specifics here, so we don't have to keep them in sync if `typescript/SKILL.md` changes?

```suggestion
Canonical source: `.cursor/skills/opentrons-typescript/SKILL.md`
```

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## 5. Component API Design (Props Extension)
+
+Standard HTML attributes must not be manually replicated. Components that wrap native HTML elements must explicitly extend native attributes to maintain web standards and prevent configuration creep.
+
+- **Anti-Pattern (Manual Shadowing)**: Do not manually type ubiquitous props like `onClick`, `className`, `id`, or `style` in custom prop interfaces.
+- **Enforced Pattern**: Always extend `React.ComponentPropsWithoutRef<'tag'>` (or `React.ComponentPropsWithRef<'tag'>` if forwarding refs).
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:37]: I'm not sure I agree with this advice.

Sometimes, it's good encapsulation when a custom component limits its interface to a small subset of what the underlying native HTML element supports.

For example, if we have a custom `<FileInput>` component, we probably _do_ want people to expose `onClick`, but we probably _don't_ want to expose `type`. We don't want someone to be able to override `type="file"` to `type="checkbox"`.

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+organisms → pages, the `no-imports-up-the-tree-of-life` ESLint rule). Cite the exact
+violated rule from that file in your `Details` — do not re-derive the boundary from memory.
+
+## 2. CSS Modules & Styles
+
+Canonical source: `.cursor/skills/css-modules/SKILL.md`.
+Open it and check file naming, class naming, design-token usage, and Stylelint rules
+against that file. Do not restate these rules here — they will drift out of sync.
+
+## 3. Web Accessibility (a11y)
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:18]: Should we move this section into `typescript/SKILL.md` so it's followed by all agent-written code, not just when we manually trigger `/grill-frontend-code`?

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md (hunk 4 above)
Comment (SyntaxColoring) [reply .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:37]: If we do keep this advice, though, can we move it somewhere like `typescript/SKILL.md` so it's followed by all agent-written code, not just when we manually trigger `/grill-frontend-code`?

File: .cursor/skills/grill-frontend-code/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+_(Optional)_ Run the relevant Vitest test suite for the modified files to ensure existing logic is unimpaired. Incorporate any discovered linter errors or test failures directly into your final output findings.
+
+3. **Evaluate against Reference** — Open `review-reference.md` and `GLOSSARY.md` via their context pointers. Evaluate every modified or added line, specifically auditing for:
+
+- **Prop Shadowing** in new component interfaces.
+- **Leak** across Atomic Design layers or style boundaries.
+- **Blind Spot** occurrences in interactive or dynamic elements.
+- **Drift** from core Opentrons patterns or design tokens.
+- **Unprotected** components or logic that lack proper sibling tests or edge-case coverage.
+- **Over-baked** components that choose speculative over-engineering over tight, modular separation.
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/SKILL.md:36]: Can we avoid mentioning specifics here so we don't need to duplicate between `review-reference.md` and `SKILL.md`?

```suggestion
3. **Evaluate against Reference** — Evaluate every modified or added line against `review-reference.md` and `GLOSSARY.md`.
```

File: .cursor/skills/grill-frontend-code/GLOSSARY.md
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/GLOSSARY.md:1]: Nitpick: Should we have consistent filename capitalization between `GLOSSARY.md` and `review-reference.md`?

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+Opentrons interfaces are used in physical lab environments. Accessibility is functional reliability.
+
+- **Interactivity**: Any clickable element that is not a semantic `<button>` or `<a>` must be rejected. Do not allow `onClick` on a raw `<div>` without appropriate `role` and `tabIndex`.
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:22]: > Any clickable element that is not a semantic `<button>` or `<a>` must be rejected.

Or `<input>`/`<select>`/`<option>`/`<summary>`/`<details>`, right? (And maybe more.)

> Do not allow `onClick` on a raw `<div>` without appropriate `role` and `tabIndex`.

Note that even with `role`+`tabIndex`+`onClick`, Enter/Spacebar won't work.

-----

What about making this the advice:

* **Interactivity**: Interactive elements must be visible to accessibility tools and must be usable with the keyboard. To accomplish this, prefer native HTML elements like `<button>`, `<input>`, and `<a>` instead of, for example, raw `<div>`s.

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+Canonical source: `.cursor/skills/opentrons-typescript/SKILL.md` § "Atomic Design Hierarchy".
+Open it and check the diff against the layer-import rules there (atoms → molecules →
+organisms → pages, the `no-imports-up-the-tree-of-life` ESLint rule). Cite the exact
+violated rule from that file in your `Details` — do not re-derive the boundary from memory.
+
+## 2. CSS Modules & Styles
+
+Canonical source: `.cursor/skills/css-modules/SKILL.md`.
+Open it and check file naming, class naming, design-token usage, and Stylelint rules
+against that file. Do not restate these rules here — they will drift out of sync.
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:16]: Can we avoid mentioning these specifics here, so we don't have to keep them in sync if `css-modules/SKILL.md` changes?

```suggestion
Canonical source: `.cursor/skills/css-modules/SKILL.md`
```

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+- **Sibling Rule**: Any new component `MyComponent.tsx` or hook `useMyHook.ts` must be accompanied by a sibling `__tests__/MyComponent.test.tsx` or `MyComponent.test.ts` file within the same directory.
+- **RTL Standards**: Opentrons uses React Testing Library.
+  - Avoid testing internal state; always test observable behavior from the user's perspective (e.g., fireEvent/userEvent).
+  - Enforce explicit MSW (Mock Service Worker) handlers for any network/API layer mock, rather than manual `vi.fn()` overrides of global fetch.
+
+## 7. Modularization vs. Over-engineering (Over-baked Code)
+
+Components must be lean, single-purpose, and modular. Do not write speculative code for future feature requirements.
+
+- **The 300-Line Ceiling**: Any component file exceeding 300 lines of code (including styles and types) is an automatic trigger for structural review. Demand that the author split the file and extract sub-sections into dedicated Atoms or Molecules.
```
Comment (SyntaxColoring) [inline .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:67]: I'm not really a fan of hard limits like this 300-line ceiling. 

And I worry that encouraging people to split big files into dedicated atoms or molecules will create a bunch of one-off atoms and molecules that don't have an equivalent in our design system, which will be confusing.

I'm willing to try this to see how it goes, though.

**Review by SyntaxColoring — APPROVED** [review]
I have various suggestions, but fundamentally I think this is a good idea and this LGTM. Thanks!

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md (hunk 5 above)
Comment (koji) [reply .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:18]: this skill is to review code and testing version and Web Accessibility part doesn't support basic yet so i don't think moving this to `typescript/SKILL.md` is good.

File: .cursor/skills/grill-frontend-code/SKILL.md (hunk 6 above)
Comment (koji) [reply .cursor/skills/grill-frontend-code/SKILL.md:36]: this is on purpose to tell LLM these are important criteria for code-review.

File: .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md (hunk 9 above)
Comment (koji) [reply .cursor/skills/grill-frontend-code/REVIEW-REFERENCE.md:67]: The number is slightly higher than the value recommended by the React community. From a performance perspective, I believe having a limit is necessary.
Generally over 500 is a sign of the immediate refactoring.
